application/dash_application/utility/df_manipulation.py

```diff
@@ -58,8 +111,51 @@
     PINERY_COL.TissueType: "n",
     PINERY_COL.TissuePreparation: "Unspecified",
     PINERY_COL.GroupID: "",
-    PINERY_COL.GroupIDDescription: ""
+    PINERY_COL.GroupIDDescription: "",
+    PINERY_COL.Institute: "Unspecified"
 })
+# Cast the primary key/join columns to explicit types now that the NA values are filled in
+_pinery_samples = _pinery_samples.astype({
+    PINERY_COL.SequencerRunName: 'str',
+    PINERY_COL.LaneNumber: 'int64',
+    "IUSTag": 'str',
+    PINERY_COL.GroupID: 'str'})
+# Fill in the "Sample Type" column (Tumor/Reference/Blood/Unknown)
+_pinery_samples[sample_type_col] = _pinery_samples.apply(label_sample_type, axis=1)
+# Drop columns we definitely don't care about.
+_pinery_samples = _pinery_samples.drop(axis=1, columns=[PINERY_COL.NanodropConcentration, PINERY_COL.QubitConcentration, "RunIDandPosition", "TargetedResequencing", "TubeID", "RIN",
+"PrepKit", "PoolName", "Organism", "STRResult", "QubitConcentration", "SampleProvenanceID",
+"TissuePreparation", "TissueType", "SampleName", "RootSampleName", "Version", "NanodropConcentration",
+"TissueOrigin", "Purpose", "SequencerRunName", "SequencingParameters", "UMIs", "GroupIDDescription",
+"DV200", "CreateDate", "InstrumentName", "ParentSampleName", "SubProject", "TemplateType",
+"RunBaseMask", "RunDir", "ExternalName", "Institute", "StudyTitle", "LastModified",
+"GroupID", "SequencerRunPlatformModel", "ReceiveDate", "TissueRegion", "WorkflowType", "Skip",
+"LaneNumber", "LibrarySourceTemplateType"])
+
+unique_list = lambda vals: ", ".join(str(val) for val in sorted(set(vals)) if (val and val != "nan"))
```

Comment:
Please comment the purpose of this (what will vals be?)

Reply:
Commented